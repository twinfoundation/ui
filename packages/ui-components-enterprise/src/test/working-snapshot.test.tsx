// Copyright 2024 IOTA Stiftung.
// SPDX-License-Identifier: Apache-2.0.

import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

// Example components that work correctly for snapshot testing
const Card = ({ title, children, className = '' }: { title: string; children: React.ReactNode; className?: string }) => {
  return (
    <div className={`card ${className}`} data-testid="card">
      <h3 className="card-title">{title}</h3>
      <div className="card-content">{children}</div>
    </div>
  );
};

const Badge = ({ text, variant = 'default' }: { text: string; variant?: 'default' | 'success' | 'error' }) => {
  return (
    <span className={`badge badge-${variant}`} data-testid="badge">
      {text}
    </span>
  );
};

describe('Snapshot Testing Examples', () => {
  describe('Card Component', () => {
    it('renders basic card correctly', () => {
      const { container } = render(
        <Card title="Sample Card">
          <p>This is card content</p>
        </Card>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders card with custom className correctly', () => {
      const { container } = render(
        <Card title="Custom Card" className="custom-card">
          <Badge text="New" variant="success" />
        </Card>
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe('Badge Component', () => {
    it('renders default badge correctly', () => {
      const { container } = render(<Badge text="Default Badge" />);
      expect(container).toMatchSnapshot();
    });

    it('renders success badge correctly', () => {
      const { container } = render(<Badge text="Success" variant="success" />);
      expect(container).toMatchSnapshot();
    });

    it('renders error badge correctly', () => {
      const { container } = render(<Badge text="Error" variant="error" />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('Complex Component Combinations', () => {
    it('renders complex layout correctly', () => {
      const { container } = render(
        <div className="dashboard">
          <Card title="User Stats">
            <div className="stats">
              <Badge text="Active" variant="success" />
              <Badge text="Pending" variant="default" />
              <Badge text="Failed" variant="error" />
            </div>
          </Card>
        </div>
      );
      expect(container).toMatchSnapshot();
    });
  });
}); 