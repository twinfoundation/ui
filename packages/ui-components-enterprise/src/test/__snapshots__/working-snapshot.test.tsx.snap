// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot Testing Examples > Badge Component > renders default badge correctly 1`] = `
<div>
  <span
    class="badge badge-default"
    data-testid="badge"
  >
    Default Badge
  </span>
</div>
`;

exports[`Snapshot Testing Examples > Badge Component > renders error badge correctly 1`] = `
<div>
  <span
    class="badge badge-error"
    data-testid="badge"
  >
    Error
  </span>
</div>
`;

exports[`Snapshot Testing Examples > Badge Component > renders success badge correctly 1`] = `
<div>
  <span
    class="badge badge-success"
    data-testid="badge"
  >
    Success
  </span>
</div>
`;

exports[`Snapshot Testing Examples > Card Component > renders basic card correctly 1`] = `
<div>
  <div
    class="card "
    data-testid="card"
  >
    <h3
      class="card-title"
    >
      Sample Card
    </h3>
    <div
      class="card-content"
    >
      <p>
        This is card content
      </p>
    </div>
  </div>
</div>
`;

exports[`Snapshot Testing Examples > Card Component > renders card with custom className correctly 1`] = `
<div>
  <div
    class="card custom-card"
    data-testid="card"
  >
    <h3
      class="card-title"
    >
      Custom Card
    </h3>
    <div
      class="card-content"
    >
      <span
        class="badge badge-success"
        data-testid="badge"
      >
        New
      </span>
    </div>
  </div>
</div>
`;

exports[`Snapshot Testing Examples > Complex Component Combinations > renders complex layout correctly 1`] = `
<div>
  <div
    class="dashboard"
  >
    <div
      class="card "
      data-testid="card"
    >
      <h3
        class="card-title"
      >
        User Stats
      </h3>
      <div
        class="card-content"
      >
        <div
          class="stats"
        >
          <span
            class="badge badge-success"
            data-testid="badge"
          >
            Active
          </span>
          <span
            class="badge badge-default"
            data-testid="badge"
          >
            Pending
          </span>
          <span
            class="badge badge-error"
            data-testid="badge"
          >
            Failed
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
